packages/components/src/ui/styles/test/css.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import { css } from '@wp-g2/styles';
+import { render } from '@testing-library/react';
+
+describe( 'basic', () => {
+	test( 'should return a string', () => {
+		const style = css`
+			background: blue;
+			color: red;
+		`;
+
+		expect( typeof style ).toBe( 'string' );
+	} );
+
+	test( 'should create styles bound to a className', () => {
+		const style = css`
+			background: blue;
+			color: red;
+		`;
+
+		const { container } = render( <div className={ style } /> );
+
+		expect( container.firstChild ).toHaveStyle( {
+			background: 'blue',
+			color: 'red',
+		} );
+	} );
+
+	test( 'should create styles from an object', () => {
+		const style = css( {
+			background: 'blue',
+			color: 'red',
+		} );
+
+		const { container } = render( <div className={ style } /> );
+
+		expect( container.firstChild ).toHaveStyle( {
+			background: 'blue',
+			color: 'red',
+		} );
+	} );
+
+	test( 'should create styles from an templates string literal', () => {
+		const background = 'blue';
+		const style = css`
+			background: ${ background };
+			color: red;
+		`;
+
+		const { container } = render( <div className={ style } /> );
+
+		expect( container.firstChild ).toHaveStyle( {
+			background: 'blue',
+			color: 'red',
+		} );
+	} );
+} );
+
+describe( 'plugins', () => {
+	test( 'should render reliably in an environment with existing styles', () => {
+		// Simulate an environment with existing styles
+		const existingStyles = document.createElement( 'style' );
+		existingStyles.innerHTML = `div.box { background: green; }`;
+		document.querySelector( 'head' ).appendChild( existingStyles );
+
+		const style = css`
+			background: blue;
+		`;
+
+		const { container } = render( <div className={ `box ${ style }` } /> );
+
+		expect( container.firstChild ).toHaveStyle( `background: blue;` );
+
+		// Clean up
+		document.querySelector( 'head' ).removeChild( existingStyles );
+	} );
+
+	test( 'should automatically render rtl styles', () => {
+		// Simulate an rtl environment
+		document.documentElement.setAttribute( 'dir', 'rtl' );
```

Comment:
It might be a good idea to read the value and store it for clean up phase if any present.

Reply:
I'll give it a shot! Thank you 🙏